pytorch_lightning/trainer/trainer.py

```diff
@@ -544,10 +547,7 @@ def _pre_training_routine(self):
 
         # print model summary
         if self.is_global_zero and self.weights_summary is not None and not self.testing:
-            if self.weights_summary in ModelSummary.MODES:
-                ref_model.summarize(mode=self.weights_summary)
-            else:
-                raise MisconfigurationException("weights_summary can be None, " + ", ".join(ModelSummary.MODES))
+            ref_model.summarize(mode=self.weights_summary)
```

Comment:
@edenafek  @awaelchli  @justusschock the model summary could be another (small) component we could split out from the core lightning module and offer as a standalone utility. the constructor for the summary object could take:
- an nn.Module
- precision
- device
- mode

connecting it back into the trainer here would be easy. it's almost there now with the `ModelSummary` object

Reply:
Related idea: https://github.com/PyTorchLightning/pytorch-lightning/issues/4541

Reply:
Btw it's already a standalone utility. It can be used like this:

```python
summary = LayerSummary(model, mode=...)
print(summary)
```
As you say it could also support nn.Module.

Reply:
by standalone i mean no dependence on the lightning module. right now the precision and example inputs are looked up via attributes of `model` inside of model summary, but these could be passed in explicitly to the model summary constructor. it's not a huge deal, but something small for people to see how lightning builds reusable components